nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2019 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.sun.source.util.Trees;
+import com.uber.nullaway.fixserialization.qual.AnnotationFactory;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.lang.model.element.Element;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+/** Config class for Fix Serialization package. */
+public class FixSerializationConfig {
+
+  /**
+   * If activated, for all reporting errors, NullAway will serialize information and suggests type
+   * changes to resolve them, in case these errors could be fixed by adding a @Nullable annotation.
+   * These data will be written at output directory.
+   */
+  public final boolean suggestEnabled;
```

Comment:
Why would we ever want this to be disabled?

Reply:
I was having in mind that this package offers serialization services. So other tools can run it with different configs. (e.g. just dumping error locations.)

For our usecase, creating ```SuggestedFix``` objects are costly and in measuring effectiveness of every ```SuggestedFix``` we only need to know the number of errors not fix objects in every run.

Reply:
> I was having in mind that this package offers serialization services. So other tools can run it with different configs. (e.g. just dumping error locations.)

I think this is a bit of over-engineering.  If we never run this code with `suggestEnabled == false`, let's just get rid of the flag for now.  We can add it back later when we have a real motivating example for it.  Same is true for any other flag in this file where we are not currently using it to disable some functionality.

Reply:
Sorry, can you update the comment to describe when we run this code with `suggestEnabled == false`?

Reply:
288dfdd9e279f2e7a76525efaf5479e47b75c72b

Reply:
So if I understand correctly, the number of errors will still be serialized even if `suggestEnabled == false`, is that correct?  Does the number of errors get serialized by this new code to a file?  I still don't fully understand what exactly gets serialized if `suggestEnabled == false`.

Reply:
Hi @msridhar, if the main flag from ```ErrorProneCLIFlag``` which is ```serializeFixMetadata``` is set to ```false```, no files will be generated, however if activated, then errors will be written at the output path regardless of all other flags in ```FixSerializationConfig```. To enable writing fix suggestions, in addition to enabling ```serializeFixMetadata``` in ErrorProneCLIFlag, ```suggestEnabled``` must also be activated in ```FixSerializationConfig```. Would you please tell me if you think renaming elements below might be helpful ?

1. ```FixSerializationConfig``` to ```SerializationConfig```
2. ```serializeFixMetadata``` in ```ErrorProneCLI``` to ```serializeMetadata```

Thank you very much.

Reply:
No need to rename anything.  Just clarify the comment to say that even if `suggestEnabled == false`, information will still be serialized on the number of errors reported (and whatever other info ends up in `errors.tsv`).